naga/src/proc/constant_evaluator.rs

```diff
@@ -2096,6 +2104,182 @@ impl<'a> ConstantEvaluator<'a> {
     }
 }
 
+fn first_trailing_bit(concrete_int: ConcreteInt<1>) -> ConcreteInt<1> {
+    // NOTE: Bit indices for this built-in start at 0 at the "right" (or LSB). For example, a value
+    // of 1 means the least significant bit is set. Therefore, an input of `0x[80 00…]` would
+    // return a right-to-left bit index of 0.
+    let trailing_zeros_to_bit_idx = |e: u32| -> u32 {
+        match e {
+            idx @ 0..=31 => idx,
+            32 => u32::MAX,
+            _ => panic!("invalid `trailing_zeros` values"),
+        }
+    };
```

Comment:
suggestion: I think this could be used in the body of `first_leading_bit` too, instead of branching on `0 | -1`?

Reply:
That would work, we'd only branch once on `e.is_positive() || e == 0`.

Reply:
Just pushed up e3293f8fcb66 as a simplification, and tests still pass. According to [Godbolt](https://godbolt.org/z/YxTz4T8fv), it reduced instruction count, yay!